### PR TITLE
Sprint 4

### PR DESCRIPTION
**Pull request by buzlan**
1)Connect cart to backend
a) when i go to the cart screen , i did get query for fetch items from cart 
Screenshot_1636624678
b) i can change quantity, and total sum will change 
Screenshot_1636624738
c) if quantity of the item ===1 and i click minus quantity, item will remove 
Screenshot_1636624794
d) if i click at the trash icon on the top right corner => all items will remove 
Screenshot_1636624832
Screenshot_1636624870
e) if i add dish without some ingredients i will see him in the cart with some comment 
Screenshot_1636625484
Screenshot_1636625489
Screenshot_1636625491
Screenshot_1636625495
f) Also, i can change ingredients from the cart 
Screenshot_1636625531
Screenshot_1636625536
2)
a) if I click next Btn at the cart screen i will navigate to OrderType screen
Screenshot_1636625554
Screenshot_1636625592
b) if i click "Бронирование стола" i will navigate to reserveTableScreen
Screenshot_1636625647
c)if i click at the data i will see calendar, and i can choose some date :
Screenshot_1636625690
Screenshot_1636625694
Screenshot_1636625696
d) if i click at time , i can choose time 
Screenshot_1636627966
Screenshot_1636627970
Screenshot_1636627973
e) if i click "Забронировать стол на " i will see dropdown list 
Screenshot_1636628015
Screenshot_1636628018
f) if i select ll fields and set checkbox button will be able to click
Screenshot_1636628052

### CODE REVIEW

File: src/Navigation/order.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            fontSize: 20,
+            fontFamily: "Roboto",
+            fontWeight: "600",
+          },
+        }}
+      />
+      <Stack.Screen
+        name="Confirmation"
+        component={confirmationScreen}
+        options={{
```
Comment (BakrID) [inline src/Navigation/order.tsx:59]: Move options object to function and pass parameters which can be changed

File: src/Screens/DishScreen/DishPage.tsx [hunk 2]
```diff
@@ -6,19 +6,21 @@ import { connect } from "react-redux";
...
+import { addItemToCart } from "../../services/store/cartStore/thunks/thunks";
 
-const DishPage = ({ route, addItemToCart }) => {
+const DishPage = ({ route, addOneItemToCart }) => {
   const [modalVisible, setModalVisible] = useState(false);
   const item = route.params.dishDetails;
   const [dish, setDish] = useState({ ...item, excludedIng: [] });
   const getNewIng = (newIng, excluded) => {
+    console.log("DISHING", dish.ingredients);
+    console.log("NEWING", newIng);
```
Comment (BakrID) [inline src/Screens/DishScreen/DishPage.tsx:19]: remove all logs

File: src/Screens/OrderType/orderTypeScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        </Text>
+      </TouchableOpacity>
+
+      <TouchableOpacity>
+        <Text style={{ fontSize: 25, fontFamily: "Roboto", fontWeight: "500" }}>
+          Доставка заказа
+        </Text>
+      </TouchableOpacity>
+
+      <TouchableOpacity>
```
Comment (BakrID) [inline src/Screens/OrderType/orderTypeScreen.tsx:30]: Move all of these buttons to one component and provide onPress, title

File: src/Screens/RegistrationScreens/loginScreen.tsx [hunk 4]
```diff
@@ -20,7 +20,10 @@ export const LoginScreen = ({ navigation }) => {
       <View style={formStyles.wrapper}>
         <Text style={stylesLoginForm.textLogo}>Вход</Text>
         <Formik
-          initialValues={{ email: "test@mail.com", password: "qwe2Fqwe" }}
+          initialValues={{
+            email: "igorbuzlanov44@gmail.com",
+            password: "qwe2Fqwe",
```
Comment (BakrID) [inline src/Screens/RegistrationScreens/loginScreen.tsx:25]: Don't push any creds to repo

File: src/Screens/TabsScreens/basketScreen.tsx [hunk 5]
```diff
@@ -1,29 +1,51 @@
...
 interface Props extends BottomTabScreenProps<ParamListBase> {}
 
-const BasketScreen: React.FC<Props> = ({ cart, totalSum }) => {
+const BasketScreen: React.FC<Props> = ({
+  cart,
+  totalSum,
+  navigation,
+  fetchAllCartItems,
+}) => {
+  console.log("NAVIGATIONBASKET", navigation);
```
Comment (BakrID) [inline src/Screens/TabsScreens/basketScreen.tsx:19]: Logs should be removed

File: src/Screens/TabsScreens/basketScreen.tsx [hunk 6]
```diff
@@ -1,29 +1,51 @@
...
+          navigation={navigation}
+        />
       ) : (
         <EmptyCart />
       )}
     </View>
   );
 };
 const mapStateToProps = (state) => {
+  console.log(state);
```
Comment (BakrID) [inline src/Screens/TabsScreens/basketScreen.tsx:38]: logs should be

File: src/components/CartItemsScreen.tsx [hunk 7]
```diff
@@ -6,12 +6,19 @@ import FAIcon from "react-native-vector-icons/FontAwesome";
...
 export const deviceWidth = Dimensions.get("window").width;
 
-const CartItemsScreen = ({ removeAllFromCart, cartItems, totalSum }) => {
+const CartItemsScreen = ({
+  removeAllFromCart,
+  cartItems,
+  totalSum,
+  navigation,
+}) => {
+  console.log("NAVIGATION", navigation);
```
Comment (BakrID) [inline src/components/CartItemsScreen.tsx:20]: logs should

File: src/components/OneItem.tsx [hunk 8]
```diff
@@ -1,38 +1,56 @@
...
+import { updateItemsFromCart } from "../actions/cart";
 import { ModalComponent } from "./Modal";
+import {
+  deleteItem,
+  updateIngredientsItem,
+  updateQuantityItem,
+} from "../services/store/cartStore/thunks/thunks";
+
 const OneItem = (props) => {
+  console.log("One Item", props.item);
```
Comment (BakrID) [inline src/components/OneItem.tsx:18]: logs

File: src/services/store/cartStore/reducers/cart.tsx [hunk 9]
```diff
@@ -16,11 +18,11 @@ const checkIngEquality = (a, b) => {
 };
 
 const removeItem = (state, action) => {
-  let copiedState = { ...state };
-  const index = copiedState.cartItems.findIndex(
-    ({ id }) => id === action.data.id
-  );
-  let updatedCartItems = produce(copiedState, (draft) => {
+  console.log("STATEREMOVE", state);
+  console.log("ACTIONREMOVE", action);
```
Comment (BakrID) [inline src/services/store/cartStore/reducers/cart.tsx:22]: .

File: src/services/store/cartStore/reducers/cart.tsx [hunk 10]
```diff
@@ -29,15 +31,17 @@ const removeItem = (state, action) => {
   };
 };
 
-const changeQuantity = (state, action, modifier) => {
+const changeQuantity = (state, action) => {
+  console.log("STATE", state);
+  console.log("ACTION", action);
```
Comment (BakrID) [inline src/services/store/cartStore/reducers/cart.tsx:36]: .

File: src/services/store/cartStore/reducers/cart.tsx [hunk 11]
```diff
@@ -70,22 +74,19 @@ const InitialState = {
   totalSum: 0,
 };
 
-const addToCart = (state, action, quantity = 1) => {
+const addToCart = (state, action) => {
+  console.log("ACTIONA", action);
```
Comment (BakrID) [inline src/services/store/cartStore/reducers/cart.tsx:78]: .

File: src/services/store/cartStore/reducers/cart.tsx [hunk 12]
```diff
@@ -99,15 +100,24 @@ const addToCart = (state, action, quantity = 1) => {
 
 export const Cart = (state = InitialState, action) => {
   switch (action.type) {
+    case ADD_ALL_ITEMS_TO_CART:
+      console.log(action, "ADD_ALL_ITEMS_TO_CART");
```
Comment (BakrID) [inline src/services/store/cartStore/reducers/cart.tsx:104]: .

**Review by BakrID — CHANGES_REQUESTED** [review]
fix all comments

File: src/services/store/cartStore/reducers/cart.tsx (hunk 12 above)
Comment (buzlan) [reply src/services/store/cartStore/reducers/cart.tsx:104]: done

File: src/services/store/cartStore/reducers/cart.tsx (hunk 11 above)
Comment (buzlan) [reply src/services/store/cartStore/reducers/cart.tsx:78]: done

File: src/services/store/cartStore/reducers/cart.tsx (hunk 10 above)
Comment (buzlan) [reply src/services/store/cartStore/reducers/cart.tsx:36]: done

File: src/services/store/cartStore/reducers/cart.tsx (hunk 9 above)
Comment (buzlan) [reply src/services/store/cartStore/reducers/cart.tsx:22]: done

File: src/components/OneItem.tsx (hunk 8 above)
Comment (buzlan) [reply src/components/OneItem.tsx:18]: done

File: src/components/CartItemsScreen.tsx (hunk 7 above)
Comment (buzlan) [reply src/components/CartItemsScreen.tsx:20]: done

File: src/Screens/TabsScreens/basketScreen.tsx (hunk 6 above)
Comment (buzlan) [reply src/Screens/TabsScreens/basketScreen.tsx:38]: done

File: src/Screens/TabsScreens/basketScreen.tsx (hunk 5 above)
Comment (buzlan) [reply src/Screens/TabsScreens/basketScreen.tsx:19]: done

File: src/Screens/RegistrationScreens/loginScreen.tsx (hunk 4 above)
Comment (buzlan) [reply src/Screens/RegistrationScreens/loginScreen.tsx:25]: its for testing

File: src/Screens/DishScreen/DishPage.tsx (hunk 2 above)
Comment (buzlan) [reply src/Screens/DishScreen/DishPage.tsx:19]: done

File: src/Navigation/order.tsx (hunk 1 above)
Comment (buzlan) [reply src/Navigation/order.tsx:59]: done

File: src/Screens/OrderType/orderTypeScreen.tsx (hunk 3 above)
Comment (buzlan) [reply src/Screens/OrderType/orderTypeScreen.tsx:30]: done

**Review by BakrID — APPROVED** [review]

**Pull request merged**